Sources/DatadogCrashReporting/DDCrashReportingPlugin.swift

```diff
@@ -4,32 +4,61 @@
  * Copyright 2019-2020 Datadog, Inc.
  */
 
+import Datadog
 import Foundation
-import CrashReporter
-
-public class DDCrashReportingPlugin {
-    private static var sharedPLCrashReporter: PLCrashReporter?
-
-    public init?() {
-        DDCrashReportingPlugin.sharedPLCrashReporter = PLCrashReporter(
-            configuration: PLCrashReporterConfig(
-                signalHandlerType: .BSD,
-                symbolicationStrategy: .all
-            )
-        )
+
+/// The implementation of `Datadog.DDCrashReportingPluginType`.
+/// Pass its instance as the crash reporting plugin for Datadog SDK to enable crash reporting feature.
+@objc
+public class DDCrashReportingPlugin: NSObject, DDCrashReportingPluginType {
+    static var thirdPartyCrashReporter: ThirdPartyCrashReporter?
+
+    // MARK: - Initialization
+
+    override public convenience init() {
+        self.init { try PLCrashReporterIntegration() }
+    }
+
+    internal init(thirdPartyCrashReporterFactory: () throws -> ThirdPartyCrashReporter) {
+        DDCrashReportingPlugin.enableOnce(using: thirdPartyCrashReporterFactory)
+    }
+
+    private static func enableOnce(using thirdPartyCrashReporterFactory: () throws -> ThirdPartyCrashReporter) {
+        if thirdPartyCrashReporter == nil {
+            do {
+                thirdPartyCrashReporter = try thirdPartyCrashReporterFactory()
+            } catch {
+                consolePrint("🔥 DatadogCrashReporting error: failed to enable crash reporter: \(error)")
+            }
+        }
     }
 
-    // TODO: RUMM-956 Revamp this by shaping the final API
-    public func testIfItWorks() {
+    // MARK: - DDCrashReportingPluginType
+
+    public func readPendingCrashReport(completion: (DDCrashReport?) -> Bool) {
+        guard let crashReporter = DDCrashReportingPlugin.thirdPartyCrashReporter,
+              crashReporter.hasPendingCrashReport() else {
+            _ = completion(nil)
+            return
+        }
+
         do {
-            guard let plCrashReporter = DDCrashReportingPlugin.sharedPLCrashReporter else {
-                print("🔥 Failed to instantiate `PLCrashReporter`")
-                return
+            let crashReport = try crashReporter.loadPendingCrashReport()
+            let wasProcessed = completion(crashReport)
+
+            if wasProcessed {
+                try? crashReporter.purgePendingCrashReport()
             }
-            try plCrashReporter.enableAndReturnError()
-            print("✅ Succeded with enabling `PLCrashReporter`")
         } catch {
-            print("🔥 Failed to enable `PLCrashReporter`: \(error)")
+            _ = completion(nil)
+            consolePrint("🔥 DatadogCrashReporting error: failed to load crash report: \(error)")
         }
     }
 }
+
+// MARK: - Utils
+
+/// Function printing `String` content to console.
+internal var consolePrint: (String) -> Void = { content in
```

Comment:
What's the added value compared to a straight call to `print()`?

Reply:
We use this pattern in `Datadog` SDK to mock the `consolePrint` in tests and add assertions on user-facing messages.

Reply:
I see:
https://github.com/DataDog/dd-sdk-ios/blob/119329d1209e97d5fbfc1c0be446f1d713f9a522/Tests/DatadogTests/Datadog/Mocks/CoreMocks.swift#L744-L757
This is clever 👍